### ISSUE TITLE
Padding is included in the computation of loss

### ISSUE THREAD

**Issue by guyang3532** [huggingface/transformers#25752]
### Description

I found the padding part in the logits (which is from embedding padding) is included in computation of loss in question answering bert:
https://github.com/huggingface/transformers/blob/main/src/transformers/models/bert/modeling_bert.py#L1876

Is it reasonable?

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Reproduced with:
`torchrun --nproc_per_node=1 examples/onnxruntime/training/question-answering/run_qa.py --model_name_or_path deepset/minilm-uncased-squad2 --do_train --do_eval --per_device_train_batch_size 16 --num_train_epochs 3 --overwrite_output_dir --output_dir ./outputs/ --per_device_eval_batch_size 16 --seed 1337 --fp16 False --skip_memory_metrics False --dataset_name squad_v2 --learning_rate 2e-5 --remove_unused_columns False --version_2_with_negative --optim adamw_ort_fused --report_to none`

### Expected behavior

Is it more reasonable to pass only valid sequence length part of logits to loss compuation with something like:
`start_loss = loss_fct(start_logits[:valid_sequence_length], start_positions)`

**Comment by ArthurZucker** [comment]
Hey! You should ask this question on [the forum](https://discuss.huggingface.co/), as this is more related to your understanding of the code 😉
